libraries/AP_Compass/AP_Compass_Backend.h

```diff
@@ -50,7 +50,7 @@ class AP_Compass_Backend
       calibration values invalid then this number must be changed.
      */
     enum DevTypes {
-        DEVTYPE_HMC5883 = 0x01,
+        DEVTYPE_HMC5883 = 0x07,
```

Comment:
we may want to keep the old value around to avoid people wanting to "fill in unused values" for new devices.